src/python/turicreate/toolkits/image_classifier/image_classifier.py

```diff
@@ -250,29 +251,74 @@ def create(
         raise TypeError("Unrecognized value for 'validation_set'.")
 
     if feature is None:
-        feature = _tkutl._find_only_image_column(dataset)
+        try:
+            feature = image_analysis.find_only_image_extracted_features_column(dataset, model)
+            feature_type = "extracted_features_array"
+        except:
+            feature = None
+
+        if feature is None:
+            try:
```

Comment:
Why not do all this in the `except` block above?

Reply:
Basically that part of the code is responsible for figuring out which column is a feature column in case the feature column in not passed by the user. 
And in that case we first see if there is a deep feature column in the sframe and if yes, we use it and if it's not found we go ahead and look for Image column. 
The try block will fail for two reasons: 
1.  There is no deep feature column
2. More than one deep feature columns were present in the sframe. (This I am assuming is by design because we throw a similar error even when more than one Image columns were found in the dataset.)
And when the `try` and `except` fail that means one of the above two cases would have happened and in that case we ignore the exception and we move ahead to see if we can find a Image column in the sframe dataset.
So this choice was by design. Please let me know if I am missing something here.